src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.LazyState.cs

```diff
@@ -22,6 +21,15 @@ internal sealed class State
             internal readonly ImmutableDictionary<SyntaxTree, ImmutableArray<LoadDirective>> LoadDirectiveMap;
             internal readonly ImmutableDictionary<string, SyntaxTree> LoadedSyntaxTreeMap;
             internal readonly ImmutableDictionary<SyntaxTree, Lazy<RootSingleNamespaceDeclaration>> RootNamespaces;
+
+            /// <summary>
+            /// The last fully computed member names for all the types (in lexicographic order) for this file. Member
```

Comment:
>in lexicographic order [](http://example.com/codeflow?start=72&length=22)

Why is this important? Do ```ImmutableDictionary``` and ```ImmutableSegmentedHashSet``` provide any ordering guarantees?

Reply:
It's important so that when we're walking the tree and generating the next SingleTypeDecls, that we can easily look back in this list to determine which set to be currently looking at to reuse.  I can doc this. 

Neither the dictionary nor hash provide ordering guarantees.

Reply:
Please clearly document which specific collection is ordered this way. There are at least three of them here.

Reply:
Done.